lib/videojs-playlist-ui.js

```diff
@@ -204,13 +204,20 @@ class PlaylistMenu extends Component {
       this.update();
     });
 
-    // keep track of whether an ad is playing so that the menu
+    // Keep track of whether an ad is playing so that the menu
     // appearance can be adapted appropriately
     player.on('adstart', () => {
       this.addClass('vjs-ad-playing');
     });
     player.on('adend', () => {
-      this.removeClass('vjs-ad-playing');
+      if (player.currentTime() === player.duration()) {
```

Comment:
I don't know if you can guarantee this degree of precision between currentTime and duration.